src/Components/Push.tsx

```diff
@@ -0,0 +1,129 @@
+    .from('updates')
+    .select()
+    .eq('updateId', '54c9244d-a336-4200-b236-fc7cf6e26a44');
+  if (data) {
+    return data[0].updateInfo;
+  }
+}
+
+// Can use this function below or use Expo's Push Notification Tool from: https://expo.dev/notifications
+async function sendPushNotification(expoPushToken: string) {
+  const message = {
+    to: expoPushToken,
+    sound: 'default',
+    title: 'A Dummy Update',
+    body: await getDummyUpdate(),
+    data: { someData: 'goes here' },
+  };
+
+  await fetch('https://exp.host/--/api/v2/push/send', {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Accept-encoding': 'gzip, deflate',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(message),
+  });
+}
+
+async function registerForPushNotificationsAsync() {
+  let token;
+
+  if (Platform.OS === 'android') {
+    Notifications.setNotificationChannelAsync('default', {
+      name: 'default',
+      importance: Notifications.AndroidImportance.MAX,
+      vibrationPattern: [0, 250, 250, 250],
+      lightColor: '#FF231F7C',
+    });
+  }
+
+  if (Device.isDevice) {
+    const { status: existingStatus } =
+      await Notifications.getPermissionsAsync();
+    let finalStatus = existingStatus;
+    if (existingStatus !== 'granted') {
+      const { status } = await Notifications.requestPermissionsAsync();
+      finalStatus = status;
+    }
+    if (finalStatus !== 'granted') {
+      alert('Failed to get push token for push notification!');
+      return '';
+    }
+    token = await Notifications.getExpoPushTokenAsync({
+      // projectId: Constants?.expoConfig?.extra?.eas.projectId, // TODO: figure out why this isnt working
+      projectId: 'd1a810ad-7132-4890-888f-0142c444b21d',
+    });
+  } else {
+    alert('Must use physical device for Push Notifications');
+  }
+
+  return token?.data ?? '';
+}
+
+export default function Push({ session }: { session: Session }) {
+  const [expoPushToken, setExpoPushToken] = useState('');
+  const [notification, setNotification] =
+    useState<Notifications.Notification>();
+  const notificationListener = useRef<Notifications.Subscription>();
+  const responseListener = useRef<Notifications.Subscription>();
+
+  useEffect(() => {
+    registerForPushNotificationsAsync().then(async token => {
+      setExpoPushToken(token);
+
+      const { error } = await supabase
+        .from('users')
+        .upsert({ userId: session?.user.id, expo_push_token: token }); // push token is used to set up edge function
+      console.log(error);
+    });
+
+    notificationListener.current =
+      Notifications.addNotificationReceivedListener(notification => {
+        setNotification(notification);
+      });
+
+    responseListener.current =
+      Notifications.addNotificationResponseReceivedListener(response => {
+        console.log(response);
+      });
+
+    return () => {
+      Notifications.removeNotificationSubscription(
```

Comment:
what is being returned from this useEffect?

Reply:
not sure... removed it bc i just copy and pasted from the example originally